src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -46,7 +46,7 @@ internal void UpdateSortedPropertyCache(ref ReadStackFrame frame)
             // Set the sorted property cache. Overwrite any existing cache which can occur in multi-threaded cases.
             if (frame.PropertyRefCache != null)
             {
-                List<PropertyRef> cache = frame.PropertyRefCache;
+                HashSet<PropertyRef> cache = frame.PropertyRefCache;
```

Comment:
For performance, we should also implement `IEqualityComparer<T>` on `PropertyRef`..

However you don't have to do this for the PR since I need to modify\use this same code for this [issue](https://github.com/dotnet/corefx/issues/36939) and I will do it at that time.